src/common/include/SIRF/common/geometrical_info.h

```diff
@@ -0,0 +1,182 @@
+	   space.
+	*/
+	typedef Coordinate Offset;
+	/*!
+	   Spacing is the physical distance between voxels in each dimensions.
+	*/
+	typedef Coordinate Spacing;
+	/*!
+	   Size is the number of voxels in each dimension.
+	*/
+	typedef Index Size;
+	/*!
+	   Each vector in Direction tells the direction of the axis in LPS
+	   physical space.
+	*/
+	typedef tVector<num_dimensions, Coordinate> DirectionMatrix;
+	/*!
+	   Each vector in Direction tells the direction of the axis in LPS
+	   physical space.
+	*/
+	typedef tVector<num_dimensions+1, tVector<num_dimensions+1, float> >
+		TransformMatrix;
+
+
+	VoxelisedGeometricalInfo(
+		const Offset& offset, const Spacing& spacing,
+		const Size& size, const DirectionMatrix& direction);
+	virtual ~VoxelisedGeometricalInfo() {};
+
+    const Offset get_offset() const;
+    const Spacing get_spacing() const;
+    const Size get_size() const;
+    const DirectionMatrix get_direction() const;
+
+    const TransformMatrix calculate_index_to_physical_point_matrix() const;
+
+private:
+	Offset offset;
+	Spacing spacing;
+	Size size;
+	DirectionMatrix direction;
+};
+
+typedef GeometricalInfo<3, 3> GeometricalInfo3D;
+typedef VoxelisedGeometricalInfo<3> VoxelisedGeometricalInfo3D;
+typedef VoxelisedGeometricalInfo<3>::TransformMatrix TransformMatrix3D;
+
+template <int num_dimensions>
+VoxelisedGeometricalInfo<num_dimensions>::
+VoxelisedGeometricalInfo(
+	const Offset& offset, const Spacing& spacing,
+	const Size& size, const DirectionMatrix& direction)
+	:
+	offset(offset),
+	spacing(spacing),
+	size(size),
+	direction(direction)
+{}
+
+template <int num_dimensions>
+const typename VoxelisedGeometricalInfo<num_dimensions>::Offset
+VoxelisedGeometricalInfo<num_dimensions>::get_offset() const
+{
+	return offset;
+}
+
+template <int num_dimensions>
+const typename VoxelisedGeometricalInfo<num_dimensions>::Spacing
+VoxelisedGeometricalInfo<num_dimensions>::get_spacing() const
+{
+	return spacing;
+}
+
+template <int num_dimensions>
+const typename VoxelisedGeometricalInfo<num_dimensions>::Size
+VoxelisedGeometricalInfo<num_dimensions>::get_size() const
+{
+	return size;
+}
+
+template <int num_dimensions>
+const typename VoxelisedGeometricalInfo<num_dimensions>::DirectionMatrix
+VoxelisedGeometricalInfo<num_dimensions>::get_direction() const
+{
+	return direction;
+}
+
+template <int num_dimensions>
+const typename VoxelisedGeometricalInfo<num_dimensions>::TransformMatrix
+VoxelisedGeometricalInfo<num_dimensions>::
+calculate_index_to_physical_point_matrix() const
+{
+	TransformMatrix index_to_physical_point_matrix;
+	for (unsigned int j = 0; j<num_dimensions; j++) {
+		for (unsigned int i = 0; i<num_dimensions; i++) {
+			// set cosines
+			index_to_physical_point_matrix[j][i] =
+				direction[j][i] * spacing[j];
+			//std::cout << direction[j][i] << ' ' << spacing[j] << '\n';
+		}
+		// set translations
+        // TODO This will only work for diagonal matrices i.e.,:
```

Comment:
not sure what you mean here. Are you talking about the order of the multiplication and addition?